app/controllers/api/v1/assignments_controller.rb

```diff
@@ -0,0 +1,106 @@
+  before_action :set_assignment, only: %i[show update destroy reopen start]
+  before_action :check_access, only: %i[update destroy reopen]
+  after_action :check_reopening_status, only: [:update]
+
+  # GET /api/v1/groups/:group_id/assignments
+  def index
+    # checks if current_user has access to group contents
+    authorize @group, :show_access?
+
+    @assignments = paginate(@group.assignments)
+    @options = { links: link_attrs(@assignments, api_v1_group_assignments_url(@group.id)) }
+    render json: Api::V1::AssignmentSerializer.new(@assignments, @options)
+  end
+
+  # GET /api/v1/assignments/:id
+  def show
+    authorize @assignment, :show?
+    render json: Api::V1::AssignmentSerializer.new(@assignment)
+  end
+
+  # POST /api/v1/groups/:group_id/assignments
+  def create
+    @assignment = @group.assignments.new(assignment_create_params)
+    authorize @assignment, :admin_access?
+    @assignment.status = "open"
+    @assignment.deadline = Time.zone.now + 1.year if @assignment.deadline.nil?
+    @assignment.save!
+    render json: Api::V1::AssignmentSerializer.new(@assignment), status: :created
+  end
+
+  # PATCH /api/v1/assignments/:id
+  def update
+    @assignment.update!(assignment_update_params)
+    if @assignment.update(assignment_update_params)
+      render json: Api::V1::AssignmentSerializer.new(@assignment), status: :accepted
+    else
+      invalid_resource!(@assignment.errors)
+    end
+  end
+
+  # DELETE /api/v1/assignments/:id
+  def destroy
+    @assignment.destroy!
+    render json: {}, status: :no_content
+  end
+
+  # GET /api/v1/assignments/:id/reopen
+  def reopen
+    if @assignment.status != "open"
+      @assignment.status = "open"
+      @assignment.deadline = Time.zone.now + 1.day
+      @assignment.save!
+      render json: { "message": "Assignment has been reopened!" }
+    else
+      api_error(status: 409, errors: "Project is already opened!")
+    end
+  end
+
+  # GET /api/v1/assignments/:id/start
+  def start
+    authorize @assignment
+    @project = @current_user.projects.new
+    @project.name = @current_user.name + "/" + @assignment.name
+    @project.assignment_id = @assignment.id
+    @project.project_access_type = "Private"
+    @project.save!
+    render json: {
+      "message": "Voila! Project set up under name #{@project.name}"
+    }
+  end
+
+  private
+
+    def set_assignment
+      @assignment = Assignment.find(params[:id])
+    end
+
+    def set_group
+      @group = Group.find(params[:group_id])
+    end
+
+    def check_reopening_status
+      @assignment.check_reopening_status
+    end
+
+    def assignment_create_params
+      params.require(:assignment).permit(
+        :name, :deadline, :description, :grading_scale, :restrictions
+      )
+    end
+
+    def assignment_update_params
+      params.require(:assignment).permit(
+        :name, :deadline, :description, :restrictions
+      )
+    end
+
+    def check_access
+      authorize @assignment, :admin_access?
```

Comment:
Why admin access is needed to create an assignment ?

Reply:
```ruby
def admin_access?
  (assignment.group&.mentor_id == user.id) || user.admin?
end
```
Group mentor or some admin can only create assignments right ? @tachyons

Reply:
That name is weird, ideally that name should be update_access or some thing similar

Reply:
Can this be updated in this PR only ? @tachyons

Reply:
@tachyons any updates here ?

Reply:
No, out of the scope for this PR